doc/content/documentation/data-structures.md

```diff
@@ -0,0 +1,223 @@
++++
+title = "Data structures"
+weight = 9
++++
+
+Phel has four main data structures. All data structures are persistent data structures. A persistent data structure preserves the previous version of itself when it is modified. Such data structures are also called immutable data structures. The difference of persistent data structures and immutable data structures is that immutable data structures copy the whole data structure, while persistent data structures share unmodified values with their previous versions.
+
+## Lists
+
+A persistent list is simple a linked list. Access or modifications on the first element is efficient, random access is not. In Phel, a list has a special meaning. They are interpreted as function calls, macro calls or special forms by the compiler.
+
+To create a list surrond the white space separeted values with parentheses or use the `list` function.
+
+```phel
+(do 1 2 3) # list with 4 entries
+(list 1 2 3) # use the list function to create a new list
+'(1 2 3) # use a quote to create a list
+```
+
+To access values in a list the functions `get`, `first`, `second`, `next` and `rest` can be used.
+
+```phel
+(get (list 1 2 3) 0) # Evaluates to 1
+(first (list 1 2 3)) # Evaluates to 1
+(second (list 1 2 3)) # Evaluates to 2
+(next (list 1 2 3)) # Evaluates to (2 3)
+(next (list)) # Evaluates to nil
+(rest (list 1 2 3)) # Evaluates to (2 3)
```

Comment:
```clojure
(next '(1))
=> nil

(rest '(1))
=> ()
```
- `next` can return nil when there is nothing
- `rest` returns always a sequence

I saw that we have this in our API already 👌🏼